build.rs

```diff
@@ -1,17 +1,51 @@
 extern crate gcc;
 
-fn main() {
-    // These next 2 lines should only be for OS X and other platforms need
-    // to do something different.
+use gcc::Config;
+use std::process::Command;
+
+fn get_llvm_output(arg: &str) -> String {
+    let res = Command::new("llvm-config").arg(arg).output().unwrap();
+    if !res.status.success() {
+        panic!("Could not run \"llvm-config {}\": {}", arg, res.status.code().unwrap());
+    }
+    String::from_utf8(res.stdout).unwrap().trim().to_string()
+}
+
+#[cfg(target_os = "macos")]
+fn get_config() -> Config {
     println!("cargo:rustc-flags=-l framework=LLDB");
     println!("cargo:rustc-flags=-L framework=/Applications/Xcode.app/Contents/SharedFrameworks");
-    gcc::Config::new()
+    let mut res = gcc::Config::new();
+    res
+        .include(env!("LLVM_ROOT").to_owned() + "/tools/lldb/include")
+        .include(env!("LLVM_ROOT").to_owned() + "/include")
+        .include(env!("LLVM_BUILD_ROOT").to_owned() + "/include");
+    res
+}
+
+#[cfg(target_os = "linux")]
+fn get_config() -> Config {
+    let llvm_headers_path = get_llvm_output("--includedir");
+    let llvm_lib_path = get_llvm_output("--libdir");
+    let llvm_version = get_llvm_output("--version").split('.').take(2).collect::<Vec<&str>>().join(".");
+    let lib_name = ["lldb-", &llvm_version].join("");
```

Comment:
This bit of code here needs a comment explaining what it is doing and why. We know here from the PR, but it should be in the code so that it is easy to understand for others in the future.